python/lsst/meas/algorithms/detection.py

```diff
@@ -779,6 +785,19 @@ def detectFootprints(self, exposure, doSmooth=True, sigma=None, clearMask=True,
 
         return results
 
+    def removeBadPixels(self, middle):
+        """Set the significance of flagged pixels to zero.
+
+        Parameters
+        ----------
+        middle : `lsst.afw.image.ExposureF`
+            Score or maximum likelihood difference image.
+            The image plane will be modified in place.
+        """
+        badPixelMask = lsst.afw.image.Mask.getPlaneBitMask(self.config.badMaskPlanes)
+        badPixels = middle.mask.array & badPixelMask > 0
+        middle.image.array[badPixels] = 0
```

Comment:
At first I was a little unnerved that you were just setting this to zero, but upon further reflection I think a whole lot more would break here if that wasn't a valid thing to do.  It might be _slightly_ better to set it to `self.config.adjustBackground`, since that claims it's a way to add some sort of fixed background to the image for debugging purposes, and then you'd want these pixels to have that background too.  But it'd probably be ever better to deprecate that config option and fix whatever test code (if any) is abusing the production task to make testing easier.  And that's of course out of scope for this ticket, so I think this is fine.